pymeasure/adapters/prologix.py

```diff
@@ -168,13 +175,20 @@ def gpib(self, address, rw_delay=None):
         rw_delay = rw_delay or self.rw_delay
         return PrologixAdapter(self.connection, address, rw_delay=rw_delay)
 
-    def wait_for_srq(self, timeout=25, delay=0.1):
+    def _check_for_srq(self, rw_delay):
+        # it was int(self.ask("++srq"))
+        self.write("++srq")
+        time.sleep(rw_delay or self.rw_delay)
+        return int(self.read())
```

Comment:
Why are we re-implementing `ask`, here? Maybe another point why ask should remain in `Adapter`?

Reply:
I do not want to have ask in the Adapter (see reasons in other comment), therefore I had to implement this special ask for a certain command.